<?xml version="1.0"?>
<robot name="Puzzlebot Jetson-Lidar Ed.">


<!-- Dummy link to fix robot chassis -->
  <link name="base_link">
  </link>
  
    <joint name="chassis_joint" type="fixed">
        <parent link="base_link"/>
        <child link="chassis"/>
        <origin xyz="0.0 0.0 0.05"/>
    </joint>

    <link name="chassis">
        <inertial>
            <origin xyz="0 0 0" rpy="0 0 0"/>
            <mass value="0.01" />
            <inertia ixx="7.58333333333e-06" ixy="0.0" ixz="0.0" iyy="7.58333333333e-06" iyz="0.0" izz="1.25e-05"/>
        </inertial>
        <collision>
            <origin rpy="0 0 0" xyz="0 0 0"/>
            <geometry>
                <mesh filename="package://puzzlebot_control/models/MCR2_1000_13_Chassis.stl"/>
            </geometry>
        </collision>
        <visual>
            <origin rpy="0 0 1.57079632679" xyz="0 0 0"/>
            <geometry>
                <mesh filename="package://puzzlebot_control/models/MCR2_1000_13_Chassis.stl"/>
            </geometry>
        </visual>
        <material name="White">
            <color rgba="0.97 0.97 0.97 1.0"/>
        </material>
    </link>

    <link name="wheel_left_link">
        <visual>
            <origin xyz="0.0 0.0 0.0" rpy="-1.57 0.0 0.0"/>
            <geometry>
                <mesh filename="package://puzzlebot_control/models/MCR2_1000_1_1_Wheel_Coupler_2.stl"/>
            </geometry>
            <material name="White">
                <color rgba="0.97 0.8 0.97 1.0"/>
            </material>
        </visual>

        <collision>
            <origin xyz="0.06 0.13 0.0" rpy="1.57 0 0"/>
            <geometry>
                <cylinder length="0.36" radius=".05" />
            </geometry>
        </collision>

        <inertial>
            <origin xyz="0.06 0.13 0.0" rpy="-1.57 0 0"/>
            <mass value="0.5"/>
            <inertia ixx="0.009" ixy="0.0" ixz="0.0" iyy="0.00" iyz="0.0" izz="0.0099"/>
        </inertial>
    </link>

    <joint name="leftWheel" type="continuous">
        <origin xyz="0.05 0.09 0.0" rpy="1.57 0.0 0.0" />
        <parent link="chassis" />
        <child link="wheel_left_link" />
        <axis xyz="0 0 1" />
    </joint>

    <link name="wheel_right_link">
        <visual>
            <origin xyz="0.0 0.0 0" rpy="1.57 0 0" />
            <geometry>
                <mesh filename="package://puzzlebot_control/models/MCR2_1000_1_1_Wheel_Coupler_2.stl"/>
            </geometry>
            <material name="White">
                <color rgba="0.97 0.8 0.97 1.0"/>
            </material>
        </visual>

        <collision>
            <origin xyz="0.06 0.13 0.0" rpy="1.57 0 0"/>
            <geometry>
                <cylinder length="0.36" radius=".05" />
            </geometry>
        </collision>

        <inertial>
            <origin xyz="0.06 0.13 0.0" rpy="1.57 0 0"/>
            <mass value="0.5"/>
            <inertia ixx="0.009" ixy="0.0" ixz="0.0" iyy="0.00" iyz="0.0" izz="0.0099"/>
        </inertial>
    </link>

    <joint name="rightWheel" type="continuous">
        <origin xyz="0.05 -0.09 0.0" rpy="1.57 0 0" />
        <parent link="chassis" />
        <child link="wheel_right_link" />
        <axis xyz="0 0 -1" />
    </joint>
 
</robot>